n3fit/src/n3fit/layers/losses.py

```diff
@@ -85,10 +81,17 @@ def call(self, y_pred, **kwargs):
         tmp = op.op_multiply([tmp_raw, self.mask])
         if tmp.shape[1] == 1:
             # einsum is not well suited for CPU, so use tensordot if not multimodel
-            right_dot = op.tensor_product(self.kernel, tmp[0, 0, :], axes=1)
-            res = op.tensor_product(tmp[0, :, :], right_dot, axes=1)
+            if len(self.kernel.shape) == 3:
+                right_dot = op.tensor_product(self.kernel[0, ...], tmp[0, 0, :], axes=1)
+                res = op.tensor_product(tmp[0, :, :], right_dot, axes=1)
+            else:
+                right_dot = op.tensor_product(self.kernel, tmp[0, 0, :], axes=1)
+                res = op.tensor_product(tmp[0, :, :], right_dot, axes=1)
```

Comment:
squeezing seems dangerous, it's not possible to squeeze a specified axis without having it error if the size of that axis is not 1, so you'd have to squeeze the whole thing. This could go wrong if there is only one data point.
I simplified it though in [1d91484](https://github.com/NNPDF/nnpdf/pull/1788/commits/1d9148403d3314d86af9a7f01386687ef52b357b)

Reply:
When you get there you are sure that it is the 1 replica case. Actually, this is not the dense but the loss, shouldn't it always have the replica axis? (thanks to what you did in the multidense)?

Reply:
Just checked this, it is only the experimental losses that do not have a replica axis (and indeed they shouldn't).

Just adding a length 1 replica axis doesn't work, since it won't match the PDF replica axis, i.e. the einsum `bri, rij, brj -> r` still doesn't work. And copying seems silly and wasteful.

How about just a comment to explain this instead? And perhaps in #1855 we can branch not on the shape of the kernel but on the type of loss (experimental vs train/val)?

Reply:
done in [49904a3](https://github.com/NNPDF/nnpdf/pull/1788/commits/49904a35885a48b395ab35fdf95a9cee786e4123)

Reply:
But couldn't we just do `[invcovmat]` for the experimental loss?

That way `right_dot = op.tensor_product(self.kernel[0, ...], tmp[0, 0, :], axes=1)` would work just the same (and similarly for the einsum)

Reply:
For the `tensor_product` it would work because in that case `1 == 1`, but for the einsum case it wouldn't because `r != 1`, all instances of `r` in the einexpr  `bri, rij, brj -> r` must be the same.